### DISCUSSION TITLE
READ_MANY에 대해 질문드립니다.

### DISCUSSION THREAD

**Discussion by Hoontou** [woowabros/nestjs-library-crud#577] [Q&A] [answered]
안녕하세요, 질문에 앞서 좋은 라이브러리 감사합니다.
![image](https://github.com/woowabros/nestjs-library-crud/assets/88626281/ccdd2a91-abd7-4632-8663-c33ec321a964)
현재 READ_MANY에서 url query로 typeorm find메서드의 where절을 명시하는건 없어진건가요?
https://youtu.be/Z0d7ZrxY-i0?si=9cza_GG_laxQq88R&t=2044
제가 설정을 잘못해서 스웨거에 안뜨는건지, 아니면 폐기된건지 궁금합니다.

where절을 명시해서 find해오려면 SEARCH만을 이용해서 가능한건가요?

**Comment by jiho-kr** [comment, marked as answer]
> READ_MANY에서 url query로 typeorm find메서드의 where절을 명시하는건 없어진건가요?

ReadMany에서는 get method로 동작하기 때문에 qyery param을 사용하여 where절 효과를 얻을 수 있고
Search에서는 post method로 동작하기 때문에 body를 사용하여 조회하도록 되어 있습니다

릴리즈 버전에 따라 swagger 명세가 변경되었다면 issue를 생성하여 버그 리포팅을 할 수 있습니다

**Reply by jiho-kr** [reply]
각 Method의 Param, Body등의 타입은 Entity에 정의된 class-validation decorator와 decorator에 설정된 groups 옵션을 통해 filter 되어 사용됩니다.

자세한 정보는 각 케이스별로 작성된 spec 파일들을 참고할 수 있습니다.

**Reply by Hoontou** [reply]
답변 너무나 감사합니다 감동입니다..

validation decorator에 그룹을 하나하나 지정하고 해결했습니다.
'그룹을 아예 지정안하고, only에 READ_MANY만을 넣고 사용해도 된다라고' 넘겨짚고 있었습니다.

너무나 감사합니다. 즐거운 주말이 되시길 간곡히 기원합니다 ㅜㅜ

---
추가로, 예제가 엄청 상세해서 참고하기에 너무나 좋습니다. 최고에요
